jwst/pipeline/calwebb_spec2.py

```diff
@@ -237,8 +238,26 @@ def process_exposure_product(
                         raise RuntimeError('Cannot determine WCS.')
 
         # Steps whose order is the same for all types of input.
+
+        # Leakcal subtraction (imprint)  occurs before background subtraction on a per-exposure basis.
+        # If there is only one `imprint` member, this imprint exposure is subtracted from all the
+        # science and background exposures.  Otherwise, there will be as many `imprint` members as
+        # there are science plus background members. Which imprint to subtract from which science/background
+        # is determined by the matching values of the header keyword PATT_NUM, or
+        # `DataModel.meta.dither.position_number`.
+
+        # find the position number of all the leak cal observations
+        imprint_pos_no = self._imprint_pos_no(members_by_type)
+
+        calibrated = self.imprint_subtract(calibrated, members_by_type['imprint'], imprint_pos_no)
+
+        # for each background image subtract an associated leak cal
+        for i, bkg_file in enumerate(members_by_type['background']):
```

Comment:
Why are there 2 calls to `imprint_subtract` (one in line 252 and another here in this loop)? Couldn't there just be one call to the step and it would then do the subtraction from all science and background members?

Reply:
Jonathan advised that the standard practice for a step is to operate on 1 data model. So that is why I designed it the way it is. If you prefer the imprint step to operate on science and backgrounds I can change the code.